tools/image-tag

```diff
@@ -4,6 +4,27 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
+REPO_ROOT=$(git rev-parse --show-toplevel)
+
+is_valid_semver() {
+      local version=$1
+      # regex taken from https://semver.org/
+      if [[ $version =~ ^v(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$ ]] ;then
+            return 1
+      else
+            return 0
+      fi
+}
+
+# Check if there is a VERSION file
+if [ -f "${REPO_ROOT}/VERSION" ]; then
+      VERSION=$(cat "${REPO_ROOT}/VERSION" )
+      if is_valid_semver "$VERSION"; then
+            echo "$VERSION"
+            exit 0
```

Comment:
So I believe that changing this script will update the version as expected b/c of this line:

https://github.com/grafana/tempo/blob/main/Makefile#L50

However I think it will also change the output of the script as used here:

https://github.com/grafana/tempo/blob/main/.drone/drone.yml#L12

which I'm pretty sure is how our docker images are tagged:

https://hub.docker.com/r/grafana/tempo/tags

@zalegrala do you know off the top of your head if this is how it works?

Reply:
question: is it fine if there is a separate `tools/version-tag`  shell script which is used in Makefile to update tag, while the existing `tools/image-tag` can be left unmodified for creating docker image id?

Thanks for higjlighting this. I didn't check this flow tbh.

Reply:
yeah. splitting the script into two for simplicity makes sense to me

Reply:
done